py/server/tests/test_barrage.py

```diff
@@ -0,0 +1,112 @@
+        super().setUpClass()
+        env = {"START_OPTS": "-DAuthHandlers=io.deephaven.auth.AnonymousAuthenticationHandler"}
+        env.update(dict(os.environ))
+        cls.server_proc = subprocess.Popen(["/opt/deephaven/server/bin/start"], shell=True, env=env,
+                                            stdin=subprocess.PIPE,
+                                            stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        cls.py_client_publish_table()
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.server_proc.kill()
+        super().tearDownClass()
+
+    @classmethod
+    def py_client_publish_table(cls):
+        from pydeephaven.session import Session, SharedTicket
+
+        for _ in range(30):
+            try:
+                Session("localhost", 10000)
+                break
+            except Exception as e:
+                time.sleep(1)
+        else:
+            raise RuntimeError("Cannot connect to the server")
+
+    def setUp(self) -> None:
+        from pydeephaven.session import Session, SharedTicket
+
+        self.pub_session = Session("localhost", 10000)
+        self.t = self.pub_session.empty_table(1000).update(["X = i", "Y = 2*i"])
+        self.shared_ticket = SharedTicket.random_ticket()
+        self.pub_session.publish_table(self.t, self.shared_ticket)
+
+    def tearDown(self) -> None:
+        self.pub_session.close()
+
+    def test_barrage_session(self):
+        session = barrage_session(host="localhost", port=10000, auth_type="Anonymous")
+        self.assertIsNotNone(session)
+
+        with self.assertRaises(DHError):
+            barrage_session(host="localhost", port=10000, auth_type="Basic", auth_token="user:password")
+
+    def test_subscribe(self):
+        session = barrage_session(host="localhost", port=10000, auth_type="Anonymous")
+        t = session.subscribe(ticket=self.shared_ticket.bytes)
+        self.assertEqual(t.size, 1000)
+        self.assertEqual(len(t.columns), 2)
+        sp = t.snapshot()
+        self.assertEqual(sp.size, 1000)
+        t1 = t.update("Z = X + Y")
+        self.assertEqual(t1.size, 1000)
+
+        with self.subTest("Invalid ticket"):
+            with self.assertRaises(DHError) as cm:
+                session.subscribe(ticket=self.shared_ticket.bytes + b"1")
+
+        with self.subTest("Table is closed"):
+            self.t.close()
+            with self.assertRaises(DHError) as cm:
+                session.subscribe(ticket=self.shared_ticket.bytes)
+
+        with self.subTest("publishing session is gone"):
+            self.pub_session.close()
+            with self.assertRaises(DHError) as cm:
+                session.subscribe(ticket=self.shared_ticket.bytes)
+
+    def test_snapshot(self):
+        session = barrage_session(host="localhost", port=10000, auth_type="Anonymous")
+        t = session.snapshot(self.shared_ticket.bytes)
+        self.assertEqual(t.size, 1000)
+        self.assertEqual(len(t.columns), 2)
```

Comment:
`test_subscribe` tests methods on the subscribed table.  Should the same happen here?